src/workers/_pca.js

```diff
@@ -13,7 +14,7 @@ function fetchPCsAsWasmArray() {
         return cache.reloaded.pcs;
     } else {
         let tmp = cache.raw.principalComponents({ copy: false });
-        return new scran.Float64WasmArray(tmp.length, tmp.byteOffset);
+        return new wa.Float64WasmArray(scran.wasmArraySpace(), null, tmp.length, tmp.byteOffset);
```

Comment:
Use `wa.createFloat64WasmArrayView(scran.wasmArraySpace(), tmp.length, tmp.byteOffset)`, which avoids the need for the caller to remember what the second argument should be.

Reply:
Now that I think about it, try doing this:

```
        return cache.raw.principalComponents({ copy: "view" });
```